libi2pd/util.cpp

```diff
@@ -143,7 +143,7 @@ namespace util
 	}
 
 	void SetThreadName (const char *name) {
-#if defined(__APPLE__) && !defined(__powerpc__)
+#if defined(__APPLE__) && !defined(__POWERPC__)
```

Comment:
Let's do this with lower-case and upper-case defines (3 checks).

Reply:
Or maybe move out those and do nothing inside separate `#else if`...

Reply:
@r4sas Since this is Apple-only condition, specifying that Linux definition is not used is redundant. (It won’t break anything, but it will do nothing.)
Apple used `__ppc__`, `__ppc64__` and `__POWERPC__`. All others are AIX, BeOS, Linux, *BSD etc.

Reply:
That's why I proposing to add empty block, like:
```cpp
#if defined(__APPLE__) && !defined(__powerpc__) && !defined(__POWERPC__)
		pthread_setname_np((char*)name);
#if defined(__APPLE__) && (defined(__powerpc__) || defined(__POWERPC__))
		// do nothing
#elif defined(__FreeBSD__) || defined(__OpenBSD__)
...
```

Reply:
> Or maybe move out those and do nothing inside separate `#else if`...

@r4sas Ideally we gonna need something like was done in Ruby, to accommodate 10.5 Intel, for example, where that function is absent as well.

Ref: https://github.com/ruby/ruby/blob/master/thread_pthread.c#L1854

Reply:
> It won’t break anything, but it will do nothing.

Ah, yes. There no `#else` block. In that case only `!defined(__powerpc__) && !defined(__POWERPC__)` will be enough.

Reply:
> That's why I proposing to add empty block, like:
> 
> ```c++
> #if defined(__APPLE__) && !defined(__powerpc__) && !defined(__POWERPC__)
> 		pthread_setname_np((char*)name);
> #if defined(__APPLE__) && (defined(__powerpc__) || defined(__POWERPC__))
> 		// do nothing
> #elif defined(__FreeBSD__) || defined(__OpenBSD__)
> ...
> ```

@r4sas Yeah, if do nothing is an acceptable behavior, this is certainly better than a broken fallback or using a non-existing function.

P. S. Do you have some reasons to use `__powerpc__` within Apple defines? I am not gonna insist here, but it is not needed. _Trust me, I know which macros work on Darwin PPC_ :)

Reply:
> > It won’t break anything, but it will do nothing.
> 
> Ah, yes. There no `#else` block. In that case changing to `!defined(__powerpc__) && !defined(__POWERPC__)` will be enough.

There is, and it breaks the build:
```
#elif !defined(__gnu_hurd__)
		pthread_setname_np(pthread_self(), name);
#endif
```
Sure enough, `__gnu_hurd__` is not defined on Apple, whether PowerPC or not, so this fallback gets triggered.

Reply:
> Sure enough, `__gnu_hurd__` is not defined on Apple, whether PowerPC or not, so this fallback gets triggered.

Oops, didn't notice the negative (`!`) sign :D

> @r4sas Yeah, if do nothing is an acceptable behavior, this is certainly better than a broken fallback or using a non-existing function.

In that case lets do it this way. Thread names are not so important.

Reply:
@r4sas Considering what Iain says here https://github.com/PurpleI2P/i2pd/issues/1726#issuecomment-1296353361 and that on 10.6 PPC (and 10.6.8 Rosetta) `pthread_setname_np` did not cause a build error, it will be better to make a do nothing fallback not for PPC, but rather for macOS < 10.6.

This will also fix the problem for old Intel systems, which likewise have no `pthread_setname_np`.

What do you think?

Reply:
@r4sas As you suggested, I added a block for do-nothing-on-old-Darwin case, and then anything except those gonna have `pthread_setname_np`. A bit simplified logic from Ruby.